src/clarity-icons/clarity-icons-element.ts

```diff
@@ -20,28 +22,30 @@ export function ClarityIconElement() {
     "use strict";
     return (parentConstructor as any).apply(this, arguments);
 }
-
-(ClarityIconElement as any).observedAttributes = ["shape", "size"];
+(ClarityIconElement as any).observedAttributes = ["shape", "size", "title"];
 
 ClarityIconElement.prototype = Object.create(
     HTMLElement.prototype, {constructor: {configurable: true, writable: true, value: ClarityIconElement}});
 
 ClarityIconElement.prototype.constructor = ClarityIconElement;
 
-const generateIcon = function(element: any, shape: string) {
-    shape = shape.split(/\s/)[0];
+const replaceTitleWithCustom = function(currentShapeTemplate: string, currentTitleAttrVal: string,
+                                        clrIconUniqId: string) {
+    const hasSVGtag = currentShapeTemplate.match(/<svg\b[^>]*>/i);  // will find the first occurrence of svg tag
 
-    if (shape !== element._shape) {
-        element._shape = shape;
+    if (!hasSVGtag) {
+        return currentShapeTemplate;
+    }
 
-        // shape exists in set
-        if (ClarityIconsApi.instance.has(shape)) {
-            element.innerHTML = ClarityIconsApi.instance.get(shape);
-        } else {
-            console.error(`'${shape}' is not found in the Clarity Icons set.`);
-            element.innerHTML = ClarityIconsApi.instance.get("error");
-        }
+    // if the title tag doesn't exist right after that first svg tag, we will inject a
+    if (currentShapeTemplate.indexOf("<title") > -1 && currentShapeTemplate.indexOf("</title>") > -1) {
+        // if title tag exists, remove that fist
+        currentShapeTemplate = currentShapeTemplate.replace(/<title[^>]*>([^<]+)<\/title>/i, "");
```

Comment:
Isn't it risky to nuke the title instead of replacing its content directly?
I could see it being an issue on custom SVGs the consumer adds to Clarity, they might have extra attributes on the title that we are not preserving here.

Reply:
I have addressed it in the updated version of the method.